middleware.ts

```diff
@@ -1,18 +1,17 @@
-import { getToken } from "next-auth/jwt";
-import { NextRequest, NextResponse } from "next/server";
+import { withAuth } from "next-auth/middleware";
+import { NextResponse } from "next/server";
 
-export default async function middleware(req: NextRequest) {
-  // Get the pathname of the request (e.g. /, /protected)
-  const path = req.nextUrl.pathname;
+export default withAuth(
+  function middleware(req) {
+    const { pathname } = req.nextUrl;
+    const loggedIn = !!req.nextauth;
+    if (["/register", "/login"].includes(pathname) && loggedIn) {
+      return NextResponse.redirect(new URL("/protected", req.url));
+    }
```

Comment:
This does not work for me locally for some reason <picture data-single-emoji=":confused_dog:" title=":confused_dog:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..vH2EDMa3IoGlP7GU.bt5ALAgsyRiyzEW2IYPahR2gbnQElFyqMlq1S1DCy914Xkmqau3hsfeu_EI1xXIVNYn_z1w84pOfu2y6X8AYSg49HD2rkf68FuSGn2Mss3mYToy2AWt8MV8SF3sIdg.VOocLoiJC7_QCWISfcU7sQ" alt=":confused_dog:" align="absmiddle"></picture>